src/Package/Impl/Publishing/Handlers/RmdPublishHandler.cs

```diff
@@ -29,45 +30,48 @@ internal sealed class RmdPublishHandler : IMarkdownFlavorPublishHandler {
             return true;
         }
 
-        public async Task Publish(IRSession session, string inputFilePath, string outputFilePath, PublishFormat publishFormat, Encoding encoding) {
-            // Send the markdown to R-Host for processing as a blob
-            byte[] rmdData = File.ReadAllBytes(inputFilePath);
-            var rmdBlobId = await session.SendBlobAsync(rmdData);
+        public async Task PublishAsync(IRSession session, ICoreShell coreShell,  IFileSystem fs, string inputFilePath, string outputFilePath, PublishFormat publishFormat, Encoding encoding) {
+            long rmdBlobId = 0;
+            long outputRmdBlobId = 0;
 
-            // Perform markdown rendering on the blob
-            string outputFile = Path.GetFileName(outputFilePath);
-            string format = GetDocumentTypeString(publishFormat);
-            string ext = Path.GetExtension(outputFilePath).Substring(1);
-            var publishResult = await session.EvaluateAsync($"rtvs:::rmarkdown_publish(blob_id = {rmdBlobId}, output_filename = '{outputFile}', output_format='{format}', encoding='cp{encoding.CodePage}')", REvaluationKind.Normal);
-            JObject obj = publishResult.Result as JObject;
-            JArray jsonBlobids = obj["blob.ids"] as JArray;
-            JArray jsonFileNames = obj["file.names"] as JArray;
+            try {
+                // Send the markdown to R-Host for processing as a blob
+                rmdBlobId = await session.SendFileAsBlobAsync(fs, inputFilePath);
+                if (rmdBlobId <= 0) {
```

Comment:
Does this indicate an error? I would expect the managed API to use exceptions for these.